lib/open_weather_client.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'rest-client'
+require 'rest_client_failure_response'
+require 'success_response'
+
+class OpenWeatherClient
+  def initialize
+    @api_key = ENV['API_KEY']
+    @api_url = ENV['API_URL']
+  end
+
+  def check_weather(city_id)
+    uri = URI.parse("#{@api_url}/weather?id=#{city_id}&units=metric&appid=#{@api_key}")
+
+    rescue_errors { RestClient.get(uri.to_s) }
```

Comment:
You could do:

```ruby
 rescue_errors { parse_json(RestClient.get(uri.to_s)) }
```

and always return `true` from `ok?` in the `SuccessResponse`. 
After all, it's the success response, it's always going to be OK.

That way the code is less obscure because you don't have to guess that the thing that can call a JSON exception is the call to `.new`. which would be unexpected.

Reply:
this will return a hash of weather data We need it to return a RestClient response